lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -84,19 +84,19 @@ def perform_request(url, method, path, params={}, body=nil)
     end
 
     def format_url(url, path=nil)
-
-      request_uri = url.uri
+      request_uri = url.clone
 
       if path
         # Combine the paths using the minimal # of /s
         # First, we make sure the path is relative so URI.join does
         # the right thing
         relative_path = path && path.start_with?("/") ? path[1..-1] : path
-        # because URI.join obliterates the query parameter, we need to save it
-        # and restore it after this operation
-        query = request_uri.query
-        request_uri = URI.join(request_uri, relative_path)
-        request_uri.query = query if query
+
+        if !request_uri.path
+          request_uri.path = path
+        else
+          request_uri.path = "#{request_uri.path}/#{relative_path}"
```

Comment:
I see that URI.join goes to great lengths to "work", using convert_to_uri on the first argument ,then using URI::Generic#merge on the other arguments, which does [a lot](https://docs.ruby-lang.org/en/2.1.0/URI/Generic.html#method-i-merge).

Off the top of my head I see no issues with just using normal string joining, do you think some problems could appear here?